src/binary-reader-linker.cc

```diff
@@ -314,12 +263,8 @@ Result BinaryReaderLinker::OnExport(uint32_t index,
 }
 
 Result BinaryReaderLinker::OnFunctionName(uint32_t index, StringSlice name) {
-  while (binary->debug_names.size() < index) {
-    binary->debug_names.emplace_back();
-  }
-  if (binary->debug_names.size() == index) {
-    binary->debug_names.push_back(string_slice_to_string(name));
-  }
+  assert(index < binary->debug_names.size());
```

Comment:
Actually it looks like the current doc says index should be within the function index space: https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md#function-names.

So I think its OK to error out here, but perhaps I should change to and error rather than an assert?